app/api/crud.py

```diff
@@ -60,9 +60,13 @@ async def get(
     if image_modal:
         params["image_modal"] = image_modal
 
-    for node_url in util.parse_nodes_as_list(util.NEUROBAGEL_NODES):
+    nodes_dict = util.parse_nodes_as_dict(util.NEUROBAGEL_NODES)
+    for node_url in nodes_dict.keys():
```

Comment:
you can use `.items()` here to extract keys and values transparently

Reply:
```
for node_url, node_name in nodes_dict.items():
        response = util.send_get_request(node_url + "query/", params)

        for result in response:
            result["node_name"] = node_name

        cross_node_results += response

    return cross_node_results
```

Is this what u had in mind?